variants/arduino_101/linker_scripts/flash.ld

```diff
@@ -180,6 +180,14 @@ SECTIONS
 
 	/* Data Closely Coupled Memory (DCCM) */
 /* DCCM Start */
+    .kernelmempool (NOLOAD) :
+    {
+	. = ALIGN(4);
+	__dccm_start = .;
+	*(.kernelmempool)
+	*(".kernelmempool.*")
+	__dccm_end = ALIGN(4);
```

Comment:
Note that this does not actually align the location counter-- it *calculates* the next aligned value based on the current location counter, although the counter itself remains unaligned. I think you should do something like this instead;

`. = ALIGN(4);`
`__dccm_end = .;`